### DISCUSSION TITLE
Randomization

### DISCUSSION THREAD

**Discussion by LecroixMalvais** [devcompl/TTController#128] [Idea]
Odd question but is there a way to go about randomizing aspects of a configuration? For example randomizing an RGB value in a COLOR configuration?

The only guess on the subject I have right now would be to use IPCServer triggers and a secondary program to send randomized 'trigger nows'

**Comment by MoshiMoshi0** [comment]
Not really, how would you want it to work? Randomize once per start? Or every minute?
I think this is better done with a custom effect, or a custom modifier.

**Comment by LecroixMalvais** [comment]
Truth be told, I am trying to replicate the Raindrop effect of either TT or iCue controllers.  The best ive come up with so far is 

			{ 
			"Type": "ColorSequenceEffect",
			"Config": {
			"Sequence": [{
			"TransitionTime": 500,
			"HoldTime": 700,
			"Color": {
			"PerLed": [
				[0,255,0],
				[0,0,0],
				[0,0,0],
				[0,50,0],
				[0,0,0],
				[0,0,0],
				[0,150,0],
				[0,0,0],
				[0,0,0]
				]	
			}},
				{
			"TransitionTime": 500,
			"HoldTime": 700,
			"Color": {
			"PerLed": [
				[0,150,0],
				[0,0,0],
				[0,0,0],
				[0,0,0],
				[0,255,0],
				[0,0,0],
				[0,50,0],
				[0,0,0],
				[0,0,0]
				]	
			}},	
				{
			"TransitionTime": 500,
			"HoldTime": 700,
			"Color": {
			"PerLed": [
				[0,50,0],
				[0,0,0],
				[0,0,0],
				[0,0,0],
				[0,150,0],
				[0,0,0],
				[0,0,0],
				[0,255,0],
				[0,0,0]
				]	
			}},
			{
			"TransitionTime": 500,
			"HoldTime": 700,
			"Color": {
			"PerLed": [
				[0,0,0],
				[0,0,0],
				[0,255,0],
				[0,0,0],
				[0,50,0],
				[0,0,0],
				[0,0,0],
				[0,150,0],
				[0,0,0]
				]	
			}},{
			"TransitionTime": 500,
			"HoldTime": 700,
			"Color": {
			"PerLed": [
				[0,0,0],
				[0,0,0],
				[0,150,0],
				[0,0,0],
				[0,0,0],
				[0,255,0],
				[0,0,0],
				[0,50,0],
				[0,0,0]
				]	
			}},{
			"TransitionTime": 500,
			"HoldTime": 700,
			"Color": {
			"PerLed": [
				[0,0,0],
				[0,255,0],
				[0,50,0],
				[0,0,0],
				[0,0,0],
				[0,150,0],
				[0,0,0],
				[0,0,0],
				[0,0,0]
				]	
			}},{
			"TransitionTime": 500,
			"HoldTime": 700,
			"Color": {
			"PerLed": [
				[0,0,0],
				[0,150,0],
				[0,0,0],
				[0,0,0],
				[0,0,0],
				[0,50,0],
				[0,0,0],
				[0,0,0],
				[0,255,0]
				]	
			}},{
			"TransitionTime": 500,
			"HoldTime": 700,
			"Color": {
			"PerLed": [
				[0,0,0],
				[0,50,0],
				[0,0,0],
				[0,255,0],
				[0,0,0],
				[0,0,0],
				[0,0,0],
				[0,0,0],
				[0,150,0]
				]	
			}},{
			"TransitionTime": 500,
			"HoldTime": 700,
			"Color": {
			"PerLed": [
				[0,0,0],
				[0,0,0],
				[0,0,0],
				[0,150,0],
				[0,0,0],
				[0,0,0],
				[0,255,0],
				[0,0,0],
				[0,50,0]
				]	
			}}
			],	
			"ColorGenerationMethod": "PerPort",
			"Trigger": {
			"Type": "AlwaysTrigger",
			}
		}
	},

but this is a only a rough approximation. It does not allow for the slow fade in or out of previous or leading LEDs in the rain sequence.

**Reply by MoshiMoshi0** [reply]
Is there a video somewhere of the effect you want? I dont recall thermaltake having raindrop effect.

**Reply by LecroixMalvais** [reply]
This is like trying to find a needle in a stack of slightly different needles..

**Reply by LecroixMalvais** [reply]
Found it. sort of.. 
https://ttrgbplus.thermaltake.com/
If you scroll down to the 'Light Modes' section,  one of the selectibles is "raindrop' and it, sort of shows how it behaves.  Oddly thoug if you goto the bottom of the page to "SEE TT RGB PLUS ECOSYSTEM IN ACTION"  they forget it exists.. Odd

**Reply by MoshiMoshi0** [reply]
I see, yea it could be kinda hacked with `ColorSequenceEffect`, but it wont be random. I was actually planning on adding that style of effect, but no promises when that will happen.

> It does not allow for the slow fade in or out of previous or leading LEDs in the rain sequence.

`ColorSequenceEffect` should blend between colors when transitioning between states based on `TransitionTime`, 500ms is not a lot so maybe bump it to 1000-2000ms

**Reply by LecroixMalvais** [reply]
Nice..  Thank you for the info. I've adjusted the json to have extremely low HoldTime vs a higher Transition Time and the effects if fairly close in appearance to raindrop, though obviously without true randomness.
